script/ramdisk.sh

```diff
@@ -0,0 +1,206 @@
+#!/bin/bash
+
+# set -exo pipefail
+set -eo pipefail
+
+# Preconditions
+
+function display_help() {
+    cat <<END
+Usage:
+  ./ramdisk.sh create name [size] [access]
+  ./ramdisk.sh destroy name
+
+Options:
+  name: disk name.
+  size: disk size in MiB, default is 128 MiB.
+  access: access type, use ReadWrite or ReadOnly, default is ReadWrite.
+END
+}
+
+system_name=$(uname -s)
+case "$system_name" in
+Darwin*)
+    OS_NAME="MACOS"
+    ;;
+Linux*)
+    OS_NAME="LINUX"
+    ;;
+*)
+    printf "running on unsupported os: $system_name\n"
+    exit 1
+    ;;
+esac
+
+if [[ $# -lt 1 ]]; then
+    display_help
+    exit 2
+fi
+
+case "$1" in
+create)
+    ACTION="CREATE"
+    ;;
+destroy)
+    ACTION="DESTROY"
+    ;;
+*)
+    printf "got unknown verb: '$1'\n"
+    exit 3
+    ;;
+esac
+
+DISK_NAME="${2// /}"
+[[ -z "$DISK_NAME" ]] && printf "got blank name: '$2'\n" && exit 4
+
+DISK_SIZE_MB=128
+if [[ ! -z "$3" ]]; then
+    if [[ $3 =~ ^[0-9]+$ ]]; then
+        DISK_SIZE_MB=$3
+    else
+        printf "got invalid size number: '$3'\n" && exit 5
+    fi
+fi
+
+case "$4" in
+ReadWrite)
+    ACCESS_TYPE=ReadWrite
+    ;;
+ReadOnly)
+    ACCESS_TYPE=ReadOnly
+    ;;
+"")
+    ACCESS_TYPE=ReadWrite
+    ;;
+*)
+    printf "got invalid access type: $4\n"
+    exit 6
+    ;;
+esac
+
+# macOS operations
+
+function macos_exist_ramdisk() {
+    DISK_ID=$(diskutil list | grep -e "APFS Volume" | grep -e "Volume\s*${DISK_NAME}\s*\d" | awk '{ print $7 }')
+    if [[ -z "$DISK_ID" ]]; then
+        # 1 as failure for non-existance
+        return 1
+    else
+        # 0 as ok for existance
+        return 0
+    fi
+}
+
+function macos_create_ramdisk() {
+    if macos_exist_ramdisk; then
+        printf "ramdisk '${DISK_NAME}' already exists: ${DISK_ID}\n"
+        exit 10
+    else
+        local DISK_SECTORS=$((2048 * $DISK_SIZE_MB))
+        DISK_ID=$(hdiutil attach -nomount ram://$DISK_SECTORS)
+        diskutil partitionDisk $DISK_ID 1 GPTFormat APFS "$DISK_NAME" '100%'
+        if ! macos_exist_ramdisk; then
+            printf "failed to create ramdisk '${DISK_NAME}'\n"
+            exit 11
+        fi
+
+        if [[ $ACCESS_TYPE == "ReadWrite" ]]; then
+            printf "ramdisk '${DISK_NAME}' just created: ${DISK_ID}\n"
+        elif [[ $ACCESS_TYPE == "ReadOnly" ]]; then
+            diskutil umount $DISK_ID
+            diskutil mount readOnly $DISK_ID
+            printf "read-only ramdisk '${DISK_NAME}' just created: ${DISK_ID}\n"
+        fi
+    fi
+}
+
+function macos_destroy_ramdisk() {
+    if macos_exist_ramdisk; then
+        diskutil umount $DISK_ID
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/an9an63/gut/pullRequest?prid=4922553)